model/testresult/testresult.go

```diff
@@ -160,3 +160,54 @@ func TestResultsQuery(taskIds []string, testId, testName, status string, limit,
 
 	return q
 }
+
+// TestResultsFilterSortPaginate is a query for returning test results to the taskTests GQL Query.
+func TestResultsFilterSortPaginate(taskIds []string, filter, status, sortBy string, sortDir, page, limit, execution int) ([]TestResult, error) {
+	tests := []TestResult{}
+	match := bson.M{
+		TaskIDKey:    bson.M{"$in": taskIds},
+		ExecutionKey: execution,
+	}
+	if len(status) > 0 {
+		match[StatusKey] = status
+	}
+
+	pipeline := []bson.M{
+		{"$match": match},
+	}
+
+	project := bson.M{
+		"duration":   bson.M{"$subtract": []string{"$" + EndTimeKey, "$" + StartTimeKey}},
+		TestFileKey:  1,
+		EndTimeKey:   1,
+		StartTimeKey: 1,
+		StatusKey:    1,
+		URLRawKey:    1,
+		URLKey:       1,
+		LogIDKey:     1,
+		LineNumKey:   1,
+		ExitCodeKey:  1,
+	}
+
+	pipeline = append(pipeline, bson.M{"$project": project})
+
+	if len(filter) > 0 {
+		matchFilterResult := bson.M{"$match": bson.M{"$or": []bson.M{{TestFileKey: bson.M{"$regex": filter, "$options": "i"}}, {StatusKey: bson.M{"$regex": filter, "$options": "i"}}}}}
```

Comment:
The filter param still applies to the status field, but I think you meant to remove it (if that's the case I would also rename filter to testName)